packages/vat-data/test/test-prepare-revocable.js

```diff
@@ -0,0 +1,127 @@
+// Modeled on test-revoke-heap-classes.js
+
+import { test } from './prepare-test-env-ava.js';
+
+// eslint-disable-next-line import/order
+import { M } from '@agoric/store';
+import { prepareExoClass, prepareExoClassKit } from '../src/exo-utils.js';
+import { makeScalarBigMapStore } from '../src/vat-data-bindings.js';
+import { prepareRevocableKit } from '../src/prepare-revocable.js';
+
+const UpCounterI = M.interface('UpCounter', {
+  incr: M.call()
+    // TODO M.number() should not be needed to get a better error message
+    .optional(M.and(M.number(), M.gte(0)))
+    .returns(M.number()),
+});
+
+const DownCounterI = M.interface('DownCounter', {
+  decr: M.call()
+    // TODO M.number() should not be needed to get a better error message
+    .optional(M.and(M.number(), M.gte(0)))
+    .returns(M.number()),
+});
+
+test('test revoke defineVirtualExoClass', t => {
+  const baggage = makeScalarBigMapStore('fakeRootBaggage');
+
+  const makeUnderlyingUpCounter = prepareExoClass(
+    baggage,
+    'UpCounter',
+    UpCounterI,
+    /** @param {number} x */
+    (x = 0) => ({ x }),
+    {
+      incr(y = 1) {
+        const { state } = this;
+        state.x += y;
+        return state.x;
+      },
+    },
+  );
+
+  const makeRevocableUpCounterKit = prepareRevocableKit(
+    baggage,
```

Comment:
When teaching our stuff, I hope to skip the `prepareExoClass` layer and go right to `zone`.
I can imagine you don't want this test to depend on anything that it doesn't have to, but... can folks who only want to use `zone` use `prepareRevokableKit`?

Reply:
This is an excellent point. I agree. But it requires a deeper refactoring, moving `prepareRevocable` to a package that can depend on zones. (It is currently in `vat-data` which cannot.)

Putting this PR into Draft until I do that refactor.

Reply:
@michaelfig , since I'm about to zone-ify, I added you as reviewer. Thanks.

Reply:
Once prepare-revocable.js is refactored to depend on the base-zone package, and given that we will want to migrate it to endo when we migrate base-zone to endo, where should it me moved now? https://github.com/Agoric/agoric-sdk/pull/8749 

Ok to move prepare-revocable.js to base-zone itself?

A separate question is what should it be named once it is refactored to depend on zones, and therefore not necessarily limited to the "prepare-for-durable" level of abstraction?

Reply:
I expect that the `@agoric/base-zone` (or `@endo/zone`) package should only contain the bare minimum API (objects and collections) that it currently has.  Could `prepare-revocable.js` be part of the contract utilities that `@agoric/zoe` currently exports, and moved only if/when we split `@agoric/zoe` into multiple packages?

Another option is to follow the trail blazed by the `@agoric/whenable` package I built.  I had originally considered putting Whenable functionality into `@agoric/base-zone`, but thought better of it and resisted the temptation.

Reply:
Oh, I misunderstood.

> moving `prepareRevocable` to a package that can depend on zones.

`prepareRevocable` can stay in `@agoric/vat-data` and take a `"devDependencies"` dependency on `@agoric/base-zone` (later `@endo/base-zone`), without any circular dependencies.  Neither of those packages currently depend on the other one.

Reply:
Ah, I think I've come around to putting it in `base-zone`.  I think it would fit well as a `@property {...} revocableKit` on the `@typedef {object} ExoZone` definition in `packages/base-zone/src/types.js`.

Sound good?

Reply:
> When teaching our stuff, I hope to skip the `prepareExoClass` layer and go right to `zone`.
> I can imagine you don't want this test to depend on anything that it doesn't have to, but... can folks who only want to use `zone` use `prepareRevokableKit`?

Zonified! Done.

Reply:
> A separate question is what should it be named once it is refactored to depend on zones, and therefore not necessarily limited to the "prepare-for-durable" level of abstraction?

Not yet renamed.

> Ah, I think I've come around to putting it in base-zone. I think it would fit well as a @property {...} revocableKit on the @typedef {object} ExoZone definition in packages/base-zone/src/types.js.

For now, I've moved prepare-revocable.js to zoe instead alongside prepare-ownable.js . I suggest we leave it there until base-zone itself moves to endo, at which time we can migrate prepare-revocable as well.

Reply:
I think this is done enough for the points raised in this conversation, so resolving. Please unresolve and comment further if you disagree.